api_client/python/timesketch_api_client/importer.py

```diff
@@ -108,37 +167,136 @@ def _reset(self):
         self._count = 0
         self._data_lines = []
 
-    def _upload_data(self, file_name, end_stream):
+    def _upload_data_buffer(self, end_stream):
         """Upload data to Timesketch.
 
         Args:
-            file_name: a full path to the file that is about to be uploaded.
             end_stream: boolean indicating whether this is the last chunk of
                 the stream.
         """
-        files = {
-            'file': open(file_name, 'rb')
+        if not self._data_lines:
+            return
+
+        start_time = time.time()
+        data = {
+            'name': self._timeline_name,
+            'sketch_id': self._sketch.id,
+            'enable_stream': not end_stream,
+            'index_name': self._index,
+            'events': '\n'.join([json.dumps(x) for x in self._data_lines]),
         }
+        logging.debug(
+            'Data buffer ready for upload, took {0:.2f} seconds to '
+            'prepare.'.format(time.time() - start_time))
+
+        response = self._sketch.api.session.post(self._resource_url, data=data)
+        # TODO: Add in the ability to re-upload failed file.
+        if response.status_code not in definitions.HTTP_STATUS_CODE_20X:
+            raise RuntimeError(
+                'Error uploading data: [{0:d}] {1:s} {2:s}, '
+                'index {3:s}'.format(
+                    response.status_code, response.reason, response.text,
+                    self._index))
+
+        logging.debug(
+            'Data buffer nr. {0:d} uploaded, total time: {1:.2f}s'.format(
+                self._chunk, time.time() - start_time))
+        self._chunk += 1
+        response_dict = response.json()
+        self._timeline_id = response_dict.get('objects', [{}])[0].get('id')
+        self._last_response = response_dict
+
+    def _upload_data_frame(self, data_frame, end_stream):
+        """Upload data to Timesketch.
+
+        Args:
+            data_frame: a pandas DataFrame with the content to upload.
+            end_stream: boolean indicating whether this is the last chunk of
+                the stream.
+        """
         data = {
             'name': self._timeline_name,
             'sketch_id': self._sketch.id,
             'enable_stream': not end_stream,
             'index_name': self._index,
+            'events': data_frame.to_json(orient='records', lines=True),
         }
 
-        response = self._sketch.api.session.post(
-            self._resource_url, files=files, data=data)
+        response = self._sketch.api.session.post(self._resource_url, data=data)
+        self._chunk += 1
+        # TODO: Add in the ability to re-upload failed file.
```

Comment:
+1 also make sure we set a limit on how many retries we do etc.

Reply:
yes, we will absolutely do that